apache/superset
# feat: cancel db query on stop

### SUMMARY

This is addressing #15402 

SQL Lab has Stop button but it only updates the state of superset without actually stopping the database query. Leaving the queries running in the background affect database performance. This PR is to stop running queries in the backend databases when:
* The Stop button pressed in SQL Lab
* Navigated to a new site, refreshed or closed the SQLLab browser window. This is optional and configurable per database at `Database -> Performance -> Cancel query on window unload event`.

It currently supports mysql, postgres and snowflake databases but new database support can be added easily. The PR is supporting sync and async query execution modes.

This is a refactored and extended version of the unmerged and inactive https://github.com/apache/superset/pull/7611


### TESTING INSTRUCTIONS
1. Run a long running query in one of the supported drivers. For example MySQL:  `SELECT sleep(60)`
2. Press the Stop button, close the tab or navigate to another Superset page
3. Running query should get killed automatically in the backend database. For example MySQL `SHOW PROCESSLIST` should not show the query from point 1)

### ADDITIONAL INFORMATION

- [x] Has associated issue: #15402
